File: test/language/packages/cpp/DefaultPackageTest.cpp
#include "gtest/gtest.h"

#include "DefaultPackageStructure.h"

TEST(DefaultPackageTest, defaultPackageStructure)
{
    // just test that DefaultPackageStructure is available in global namespace
    DefaultPackageStructure structure;
    structure.initialize(4);
    structure.setValue(10);
    default_package_import::top::TopStructure topStructure;
    topStructure.setType(1);
    topStructure.setData(1234);
    structure.setTopStructure(topStructure);
    ASSERT_EQ(10, structure.getValue());
    ASSERT_EQ(1, structure.getTopStructure().getType());
    ASSERT_EQ(1234, structure.getTopStructure().getData());
}

File: test/language/packages/java/DefaultPackageTest.java
import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.Test;

import java.math.BigInteger;


public class DefaultPackageTest
{
    @Test
    public void defaultPackageStructure() throws Exception
    {
        // just test that DefaultPackageStructure is available in default package
        DefaultPackageStructure structure = new DefaultPackageStructure((short)4);
        structure.setValue(BigInteger.valueOf(10));
        structure.setTopStructure(new default_package_import.top.TopStructure((short)1, 1234));
        structure.setChildStructure(new Child(0xdeadbeefL));
        assertEquals(BigInteger.valueOf(10), structure.getValue());
        assertEquals(1, structure.getTopStructure().getType());
        assertEquals(1234, structure.getTopStructure().getData());
        assertEquals(0xdeadbeefL, structure.getChildStructure().getValue());
    }
};
